### PR TITLE
send current page url to Short homepage.

### LINKED ISSUE #11: Automatically fill in long URL
## Scenario
After the user clicks on the icon of the [Chrome Extension](https://github.com/byliuyang/short-ext)
Screen Shot 2019-10-07 at 8 47 10 AM
 , the extension will grab the URL of the current page and redirect the user to the app’s home page with `&long_link=http://the_current_page_url` appended in the URL. The page automatically fill in the URL of the current page, and move the cursor to the alias text field, waiting for the user to create a the short link.

## Success criteria
The extension redirects the user to the home page with query parameter `long_link=http://the_current_page_url`appended.

## Related issue
https://github.com/byliuyang/short/issues/205

### PR DESCRIPTION
**Pull request by fxnoob**
click on extension icon to open short homepage with long_link param having current page's url as value.
resolves #11

### CODE REVIEW

File: src/background.ts [hunk 1]
```diff
@@ -52,6 +53,14 @@ class ShortExt {
             });
     }
 
+    iconClickHandler = (webUi: string) => {
+        chrome.browserAction.onClicked.addListener(tab => {
+            if(tab.hasOwnProperty("url")) {
+                chrome.tabs.create({ url: `${webUi}/?long_link=${tab.url}` }, () => {});
```
Comment (naturetime10) [inline src/background.ts]: Is this part needed?
```
() => {}
```

File: src/background.ts [hunk 2]
```diff
@@ -52,6 +53,14 @@ class ShortExt {
             });
     }
 
+    iconClickHandler = (webUi: string) => {
+        chrome.browserAction.onClicked.addListener(tab => {
+            if(tab.hasOwnProperty("url")) {
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
            if(isEmptyTab(tab)) {
              return;
            }
```

File: src/background.ts [hunk 3]
```diff
@@ -52,6 +53,14 @@ class ShortExt {
...
+                chrome.tabs.create({ url: `${webUi}/?long_link=${tab.url}` }, () => {});
+            }
```
Comment (naturetime10) [inline src/background.ts:67]:
```suggestion

```

File: src/background.ts [hunk 4]
```diff
@@ -34,6 +34,7 @@ function openTab(url: string) {
 class ShortExt {
     constructor(private apiBaseUrl: string, private webUi: string) {
         this.setupOmnibox();
+        this.iconClickHandler(webUi);
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
        this.onIconClick(webUi);
```

File: src/background.ts [hunk 5]
```diff
@@ -52,6 +53,14 @@ class ShortExt {
             });
     }
 
+    iconClickHandler = (webUi: string) => {
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
    onIconClick = (webUi: string) => {
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/background.ts [hunk 6]
```diff
@@ -52,6 +53,17 @@ class ShortExt {
...
     }
 
+    isEmptyTab(tab: chrome.tabs.Tab) {
+        return !tab.hasOwnProperty("url");
+    }
+
+    onIconClick = (webUi: string) => {
+        chrome.browserAction.onClicked.addListener(tab => {
+            if(this.isEmptyTab(tab)) {
+                chrome.tabs.create({ url: `${webUi}/?long_link=${tab.url}` });
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
                     return;
                 }
                chrome.tabs.create({ url: `${webUi}/?long_link=${tab.url}` });
```

Comment (naturetime10) [reply src/background.ts]: Please compare the conditional inside the `if` statement with the previous comments.

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/background.ts (hunk 6 above)
Comment (fxnoob) [reply src/background.ts]: oops. yes

File: src/background.ts [hunk 7]
```diff
@@ -52,6 +53,22 @@ class ShortExt {
...
+    }
+
+    onIconClick = (webUi: string) => {
+        chrome.browserAction.onClicked.addListener(tab => {
+            if(this.isEmptyTab(tab)) {
+               return;
+            }
+            if(tab.url!.includes("https://s.time4hacks.com")) {
+                return;
+            }
```
Comment (naturetime10) [inline src/background.ts:67]: May I ask what is this for? Also let's use webUi instead of hard coding URL here

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/background.ts (hunk 7 above)
Comment (fxnoob) [reply src/background.ts:67]: if the user is already on dashboard then user should not get redirected again.

**Pull request closed without merging**